From f37a2871c57468d317739e21cc747066d9ed284f Mon Sep 17 00:00:00 2001
From: Gerd Hoffmann <kraxel@redhat.com>
Date: Mon, 3 Feb 2014 16:30:29 +0100
Subject: [PATCH 05/28] usb: move usb_{hi, lo} helpers to header file.

RH-Author: Gerd Hoffmann <kraxel@redhat.com>
Message-id: <1391445032-5540-3-git-send-email-kraxel@redhat.com>
Patchwork-id: 57076
O-Subject: [RHEL-7 qemu-kvm PATCH 2/5] usb: move usb_{hi, lo} helpers to header file.
Bugzilla: 1039530
RH-Acked-by: Paolo Bonzini <pbonzini@redhat.com>
RH-Acked-by: Juan Quintela <quintela@redhat.com>
RH-Acked-by: Miroslav Rezanina <mrezanin@redhat.com>

Signed-off-by: Gerd Hoffmann <kraxel@redhat.com>
(cherry picked from commit 0b1fa34e1dd2764ee7ae2be849e1c5ba5e8724ca)
---
 hw/usb/desc.c | 10 ----------
 hw/usb/desc.h | 11 +++++++++++
 2 files changed, 11 insertions(+), 10 deletions(-)

Signed-off-by: Miroslav Rezanina <mrezanin@redhat.com>
---
 hw/usb/desc.c |   10 ----------
 hw/usb/desc.h |   11 +++++++++++
 2 files changed, 11 insertions(+), 10 deletions(-)

diff --git a/hw/usb/desc.c b/hw/usb/desc.c
index bf6c522..3e560cd 100644
--- a/hw/usb/desc.c
+++ b/hw/usb/desc.c
@@ -6,16 +6,6 @@
 
 /* ------------------------------------------------------------------ */
 
-static uint8_t usb_lo(uint16_t val)
-{
-    return val & 0xff;
-}
-
-static uint8_t usb_hi(uint16_t val)
-{
-    return (val >> 8) & 0xff;
-}
-
 int usb_desc_device(const USBDescID *id, const USBDescDevice *dev,
                     uint8_t *dest, size_t len)
 {
diff --git a/hw/usb/desc.h b/hw/usb/desc.h
index ddd3e74..81327b0 100644
--- a/hw/usb/desc.h
+++ b/hw/usb/desc.h
@@ -194,6 +194,17 @@ struct USBDesc {
 
 #define USB_DESC_FLAG_SUPER (1 << 1)
 
+/* little helpers */
+static inline uint8_t usb_lo(uint16_t val)
+{
+    return val & 0xff;
+}
+
+static inline uint8_t usb_hi(uint16_t val)
+{
+    return (val >> 8) & 0xff;
+}
+
 /* generate usb packages from structs */
 int usb_desc_device(const USBDescID *id, const USBDescDevice *dev,
                     uint8_t *dest, size_t len);
-- 
1.7.1

